cabal-install/Distribution/Client/TargetProblem.hs

```diff
@@ -0,0 +1,87 @@
+module Distribution.Client.TargetProblem
+  ( TargetProblem(..),
+    TargetProblem',
+    TargetProblemCommon(..),
+    commonTargetProblem,
+    noneEnabledTargetProblem,
+    noTargetsProblem,
+    customTargetProblem,
+  )
+where
+
+import Distribution.Client.Compat.Prelude
+import Distribution.Client.ProjectPlanning
+  ( AvailableTarget,
+  )
+import Distribution.Client.TargetSelector
+  ( TargetSelector, SubComponentTarget,
+  )
+import Distribution.Package
+  (PackageId, PackageName,
+  )
+import Distribution.Types.UnqualComponentName
+  ( UnqualComponentName,
+  )
+import Distribution.Simple.LocalBuildInfo
+  ( ComponentName(..),
+  )
+import Prelude ()
+
+-- | Target problems that occur during project orchestration.
+data TargetProblemCommon
+   = TargetNotInProject                   PackageName
```

Comment:
3 spaces,  cabal is truly diverse!

Reply:
Haha! I really wish we could just run a formatter over the whole code base (I don't care which one!) so that I can stop worrying about indentation.